gamla/immutable_set.py

```diff
@@ -0,0 +1,51 @@
+from typing import Iterable, Tuple
+
+import immutables
+
+
+# Just for typing.
+class ImmutableSet(immutables.Map):
+    pass
+
+
+def create(iterable: Iterable) -> ImmutableSet:
+    return immutables.Map(map(lambda x: (x, None), iterable))
+
+
+EMPTY: ImmutableSet = create([])
+
+
+def add(set: ImmutableSet, element) -> ImmutableSet:
+    return set.set(element, None)
+
+
+def remove(set: ImmutableSet, element) -> ImmutableSet:
+    return set.delete(element)
+
+
+def contains(set: ImmutableSet, element) -> bool:
+    return element in set
+
+
+def _find_larger_smaller(
+    set1: ImmutableSet,
+    set2: ImmutableSet,
+) -> Tuple[ImmutableSet, ImmutableSet]:
+    if len(set1) > len(set2):
+        return set1, set2
+    return set2, set1
+
+
+def union(set1: ImmutableSet, set2: ImmutableSet) -> ImmutableSet:
+    larger, smaller = _find_larger_smaller(set1, set2)
+    for element in smaller:
+        larger = add(larger, element)
```

Comment:
i did some checks and i think we should use update instead of add.
time for union of 2 sets with size 10000:
1. with a for loop of add: 0.0122661010000229
2. with update: 0.006694373999948766

Reply:
ptal

Reply:
weird - if i change the implementation my test breaks (the opposite of what you say)